builtin/providers/ultradns/resource_ultradns_dirpool.go

```diff
@@ -0,0 +1,612 @@
+										Optional: true,
+									},
+									"is_account_level": &schema.Schema{
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"ips": &schema.Schema{
+										Type:     schema.TypeSet,
+										Optional: true,
+										Set:      hashIPInfoIPs,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"start": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"cidr", "address"},
+												},
+												"end": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"cidr", "address"},
+												},
+												"cidr": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"start", "end", "address"},
+												},
+												"address": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"start", "end", "cidr"},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			// Optional
+			"ttl": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  3600,
+			},
+			"conflict_resolve": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "GEO",
+				// Valid: "GEO", "IP"
+			},
+			"no_response": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"all_non_configured": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"geo_info": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"is_account_level": &schema.Schema{
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"codes": &schema.Schema{
+										Type:     schema.TypeSet,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+								},
+							},
+						},
+						"ip_info": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"is_account_level": &schema.Schema{
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"ips": &schema.Schema{
```

Comment:
is this the same schema as the `ips` above? If so, this can be extracted to a common schema :) 

See [replication groups](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_elasticache_replication_group.go#L19) for an example - it calls a common schema and then overrides specific values when needed